app/Http/Controllers/TourController.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Tour;
+
+class TourController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $tours = Tour::oldest()->paginate(config('app.default_paginate_tour'));
+        foreach ($tours as $tour) {
+            $images = $tour->images->first();
+            if (!$images) {
+                $arr[] = 'assets/images/destinations/NotFound.png';
+            } else $arr[] = $images->url;
+        }
+
+        return view('destinations', compact('tours', 'arr'));
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        //
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        //
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        $tour = Tour::findOrFail($id);
+        if (!$tour) {
+            return redirect()->route('tour.index')->with('error', trans('messages.not_found_tour'));
+        }
+        $images = $tour->images->all();
+        if (!$images) {
+            $img_fail = '/assets/images/destinations/NotFound.png';
+            return view('tour', compact('tour', 'img_fail'));
```

Comment:
em dùng findOrFail thì sẽ ko bắt được các case này đâu. dùng try catch để handle findOrFail() nhé

Reply:
Dạ em sửa thành find rồi ạ, e cho thông báo Not Found nếu ko tìm thấy id

Reply:
code vẫn vậy mà e? em dùng try cache xử lý trường hợp này